### PR TITLE
Fix Camera_Update fake, match Play_UpdateMain & Play_DrawMain NON_MATCHINGs

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

File: src/code/z_play.c [hunk 1]
```diff
@@ -1027,17 +1029,16 @@ void Play_UpdateMain(PlayState* this) {
         }
     }
 
-    if (!sp5C || gDbgCamEnabled) {
-        s32 sp54;   // camId
-        Vec3s sp48; // InputDir
+    if ((!sp5C) || gDbgCamEnabled) {
```
Comment (engineer124) [inline src/code/z_play.c:1032]:
```suggestion
    if (!sp5C || gDbgCamEnabled) {
```
Here and other places, a lot of extra brackets seems to have been added

File: src/code/z_play.c [hunk 2]
```diff
@@ -960,31 +960,33 @@ void Play_UpdateMain(PlayState* this) {
...
-                if ((this->actorCtx.freezeFlashTimer != 0) && (this->actorCtx.freezeFlashTimer-- < 5)) {
+
+                if ((this->actorCtx.freezeFlashTimer != 0) && ((this->actorCtx.freezeFlashTimer--) < 5)) {
                     freezeFlashTimer = this->actorCtx.freezeFlashTimer;
                     if ((freezeFlashTimer > 0) && ((freezeFlashTimer % 2) != 0)) {
                         this->envCtx.fillScreen = true;
-                        this->envCtx.screenFillColor[0] = this->envCtx.screenFillColor[1] =
-                            this->envCtx.screenFillColor[2] = 150;
+                        this->envCtx.screenFillColor[0] =
+                            (this->envCtx.screenFillColor[1] = (this->envCtx.screenFillColor[2] = 150));
```
Comment (engineer124) [inline src/code/z_play.c:989]:
```suggestion
                            this->envCtx.screenFillColor[1] = this->envCtx.screenFillColor[2] = 150;
```

File: src/code/z_play.c [hunk 3]
```diff
@@ -960,31 +960,33 @@ void Play_UpdateMain(PlayState* this) {
...
-
             AnimationContext_Reset(&this->animationCtx);
             Object_UpdateEntries(&this->objectCtx);
 
             if (!sp5C && (IREG(72) == 0)) {
                 this->gameplayFrames++;
                 Rumble_SetUpdateEnabled(true);
-                if ((this->actorCtx.freezeFlashTimer != 0) && (this->actorCtx.freezeFlashTimer-- < 5)) {
+
+                if ((this->actorCtx.freezeFlashTimer != 0) && ((this->actorCtx.freezeFlashTimer--) < 5)) {
```
Comment (engineer124) [inline src/code/z_play.c:984]:
```suggestion
                if ((this->actorCtx.freezeFlashTimer != 0) && (this->actorCtx.freezeFlashTimer-- < 5)) {
```

File: src/code/z_play.c [hunk 4]
```diff
@@ -960,31 +960,33 @@ void Play_UpdateMain(PlayState* this) {
...
-            if ((gSaveContext.gameMode == GAMEMODE_NORMAL) &&
-                (((this->msgCtx.msgMode == MSGMODE_NONE)) ||
-                 ((this->msgCtx.currentTextId == 0xFF) && (this->msgCtx.msgMode == MSGMODE_TEXT_DONE) &&
-                  (this->msgCtx.textboxEndType == TEXTBOX_ENDTYPE_41)) ||
-                 ((this->msgCtx.currentTextId >= 0x100) && (this->msgCtx.currentTextId <= 0x200))) &&
+            if (((gSaveContext.gameMode == GAMEMODE_NORMAL) &&
+                 (((this->msgCtx.msgMode == MSGMODE_NONE) ||
+                   (((this->msgCtx.currentTextId == 0xFF) && (this->msgCtx.msgMode == MSGMODE_TEXT_DONE)) &&
+                    (this->msgCtx.textboxEndType == TEXTBOX_ENDTYPE_41))) ||
+                  ((this->msgCtx.currentTextId >= 0x100) && (this->msgCtx.currentTextId <= 0x200)))) &&
```
Comment (engineer124) [inline src/code/z_play.c:971]: A few extra brackets added in here

**Review by engineer124 — COMMENTED** [review]
Nice find!

File: src/code/z_play.c [hunk 5]
```diff
@@ -960,25 +960,27 @@ void Play_UpdateMain(PlayState* this) {
...
-                (((this->msgCtx.msgMode == MSGMODE_NONE)) ||
-                 ((this->msgCtx.currentTextId == 0xFF) && (this->msgCtx.msgMode == MSGMODE_TEXT_DONE) &&
-                  (this->msgCtx.textboxEndType == TEXTBOX_ENDTYPE_41)) ||
-                 ((this->msgCtx.currentTextId >= 0x100) && (this->msgCtx.currentTextId <= 0x200))) &&
-                (this->gameOverCtx.state == GAMEOVER_INACTIVE)) {
+            if ((gSaveContext.gameMode == GAMEMODE_NORMAL &&
+                 ((this->msgCtx.msgMode == MSGMODE_NONE ||
+                   ((this->msgCtx.currentTextId == 0xFF && this->msgCtx.msgMode == MSGMODE_TEXT_DONE) &&
+                    (this->msgCtx.textboxEndType == TEXTBOX_ENDTYPE_41))) ||
+                  ((this->msgCtx.currentTextId >= 0x100) && (this->msgCtx.currentTextId <= 0x200)))) &&
```
Comment (engineer124) [inline src/code/z_play.c:971]: I think you took off brackets at the wrong spots. I would just copy what was there before, there should be no diff-change here:
```c
            if ((gSaveContext.gameMode == GAMEMODE_NORMAL) &&
                (((this->msgCtx.msgMode == MSGMODE_NONE)) ||
                 ((this->msgCtx.currentTextId == 0xFF) && (this->msgCtx.msgMode == MSGMODE_TEXT_DONE) &&
                  (this->msgCtx.textboxEndType == TEXTBOX_ENDTYPE_41)) ||
                 ((this->msgCtx.currentTextId >= 0x100) && (this->msgCtx.currentTextId <= 0x200))) &&
                (this->gameOverCtx.state == GAMEOVER_INACTIVE)) {
```

File: src/code/z_play.c [hunk 6]
```diff
@@ -960,25 +960,27 @@ void Play_UpdateMain(PlayState* this) {
...
+                this->gameOverCtx.state == GAMEOVER_INACTIVE) {
                 KaleidoSetup_Update(this);
             }
 
             sp5C = (this->pauseCtx.state != 0) || (this->pauseCtx.debugEditor != DEBUG_EDITOR_NONE);
-
             AnimationContext_Reset(&this->animationCtx);
```
Comment (engineer124) [inline src/code/z_play.c:977]: I wowuld keep this space here as well

**Review by hensldm — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

File: src/code/z_play.c [hunk 7]
```diff
@@ -1028,16 +1031,15 @@ void Play_UpdateMain(PlayState* this) {
...
     if (!sp5C || gDbgCamEnabled) {
-        s32 sp54;   // camId
-        Vec3s sp48; // InputDir
+        s32 i;
 
         this->nextCamera = this->activeCamId;
-        for (sp54 = 0; sp54 < 4; sp54++) {
-            if ((sp54 != this->nextCamera) && (this->cameraPtrs[sp54] != NULL)) {
-                Camera_Update(&sp48, this->cameraPtrs[sp54]);
+        for (i = 0; i < 4; i++) {
```
Comment (AngheloAlf) [inline src/code/z_play.c:1037]:
```suggestion
        for (i = 0; i < NUM_CAMS; i++) {
```

**Review by AngheloAlf — APPROVED** [review]

**Review by isghj5 — APPROVED** [review]

**Pull request merged**